main.py

```diff
@@ -0,0 +1,118 @@
+'''
+GAME OF LIFE, a basic implementation using pygame and numpy.
+
+@author: Fendross, github page: https://github.com/Fendross
+'''
+
+import time
+import pygame as pg
+import numpy as np
+
+# Game Parameters
+BG_COLOR = (20, 20, 20) # black-ish
+GRID_COLOR = (20, 200, 20) # bright green
+DEAD_CELL_COLOR = (170, 170, 170) # gray
+ALIVE_CELL_COLOR = (255, 255, 255) # white
+CELL_SIZE = 10 # TODO: game not optimal for other values, fix it
+SLEEP_TIME = 0.01
+
+
+def update(screen, cells, size, with_progress=False):
+    '''
+    @notice: the core of the simulation, it updates the state of the cells in the grid by flipping it or maintaining the same, based on
+             the Game of Life rules.
+             Case alive: if less than 2 or more than 3 alive neighbors, the cell dies. If either 2 or 3 alive neighbors, no change.
+             Case dead: if exactly 3 alive neighbors, the cell comes to life.
+    
+    @param screen: the screen object of pygame.
+    @param cells: a 2D numpy array object, used to keep track of the cells state.
+    @param size: the size of the cells.
+    @param with_progress: gives the option to proceed with the simulation or stop it. If not explicit, set to False.
+    @return: a 2D array of updated cells, updated_cells.
+    '''
+    # An array of shape cells filled with 0.
+    updated_cells = np.zeros((cells.shape[0], cells.shape[1]))
+
+    # Iterating over every row and col of cells
+    for row, col in np.ndindex(cells.shape):
+        # Calculate the number of alive neighbor cells and subtract the contribution of the current cell, it doesn't count as a neighbor
+        alive = np.sum(cells[row - 1 : row + 2, col - 1 : col + 2]) - cells[row, col]
+
+        # Get the color of the current cell
+        if cells[row, col] == 0:
```

Comment:
NOT a problem but know that you can write this also as `color = BG_COLOR if cells[row, col] == 0 else ALIVE_CELL_COLOR `